bookdown/07-special-cluster.Rmd

```diff
@@ -0,0 +1,109 @@
+However, unlike classification problems where each observation is labeled, clustering works on data sets without true labels or class assignments.
+
+
+The package `r mlr_pkg("mlr3cluster")` extends `r mlr_pkg("mlr3")` with the following objects for cluster analysis:
+
+* `r ref("mlr3cluster::TaskClust", text = "TaskClust")` to define clustering tasks
+* `r ref("mlr3cluster::LearnerClust", text = "LearnerClust")` as base class for clustering learners
+* `r ref("mlr3cluster::PredictionClust", text = "PredictionClust")` as specialized class for `r ref("Prediction")` objects
+* `r ref("mlr3cluster::MeasureClust", text = "MeasureClust")` as specialized class for performance measures
+
+Since clustering is a type of unsupervised learning, `TaskClust` is slightly different from `TaskRegr` and `TaskClassif` objects.
+More specifically:
+
+  * `truth()` function is missing because observations are not labeled.
+  * `target` field is empty and will return `character(0)` if accessed anyway.
+
+
+### Train and Predict
+Clustering learners provide both `train` and `predict` methods.
+The analysis typically consists of building clusters using all available data.
+Nevertheless, some learners can assign new observations to existing groups with `predict`.
+However, prediction does not always make sense, e.g., for hierarchical clustering.
+Moreover, some learners estimate the probability of each observation belonging to a given cluster.
+`predict_types` field gives a list of prediction types for each learner.
+
+After training, the `model` field stores a learner's model that looks different for each learner depending on the underlying library.
+`predict` returns a `PredictionClust` object that gives a simplified view of the learned model.
+If the data given to the `predict` method is the same as the one on which the learner was trained, `predict` simply returns cluster assignments for the "training" observations.
+On the other hand, if the test set contains new data, `predict` will estimate cluster assignments for that data set (some learners do not do that, so read the documentation first).
+
+In the following example, a `r ref("mlr_learners_clust.kmeans", text = "$k$-means learner")` is applied on the `r ref("mlr_tasks_usarrests", text = "US arrest data set")`.
+The class labels are predicted and the contribution of the task features to assignment of the respective class are visualized.
+```{r 07-special-cluster-002, message=FALSE, warning=FALSE}
+library("mlr3")
+library("mlr3cluster")
+library("mlr3viz")
+
+# create an example task
+task = tsk("usarrests")
+print(task)
+autoplot(task)
+
+# create a k-means learner
+learner = lrn("clust.kmeans")
+
+# assigning each observation to one of the two clusters (default in clust.kmeans)
+learner$train(task)
+learner$model
+
+# make "predictions" for the same data
+prediction = learner$predict(task)
+autoplot(prediction, task)
+```
+
+
+### Measures
```

Comment:
This section is a bit weird in that in only mentions one measure, which is only shown through visualization (the next section). I would mention at least one other measure here, and something that gives a numeric quantity, similar to e.g. accuracy in supervised problems.